test/new-e2e/pkg/components/kubernetes_agent.go

```diff
@@ -0,0 +1,18 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package components
+
+import (
+	"github.com/DataDog/test-infra-definitions/components/datadog/agent"
+)
+
+// KubernetesAgent is an agent running in a Kubernetes cluster
+type KubernetesAgent struct {
+	agent.KubernetesAgentOutput
+
+	// Client cannot be initialized inline as it requires other information to create client
+	// Client agentclient.Agent // Do we need to communicate with the Agent, currently tests do not
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Fair point, don't think normally customers interact with the agent this way.

Reply:
I left it commented, because I feel like we could need to be able to execute agent command in the pod the agent is running. But I did not see a test doing it yet